### PR TITLE
DOC-3785 updated clients landing page

### PR DESCRIPTION
**Pull request by andy-stark-redis**
A few small tasks here:

- Simplified the text on the clients landing page
- Added a list of "minor" clients
- Moved Go from the sidebar to the minor clients list
- Reordered the pages to put Python at the top

### CODE REVIEW

File: content/develop/connect/clients/_index.md [hunk 1]
```diff
@@ -15,20 +15,29 @@ title: Connect with Redis clients
...
-* [Redis OM .NET]({{< relref "/integrate/redisom-for-net" >}})
-* [Redis OM Node]({{< relref "/integrate/redisom-for-node-js/" >}})
-* [Redis OM Python]({{< relref "/integrate/redisom-for-python/" >}})
-* [Redis OM Spring]({{< relref "/integrate/redisom-for-java/" >}})
-
-<hr>
\ No newline at end of file
+Use the Redis client libraries to connect to Redis servers from
+your own code. We currently support and document client libraries
+for 4 main languages:
```
Comment (dwdougherty) [inline content/develop/connect/clients/_index.md:20]: Two style guide violations: (1) "...currently...", see [here](https://developers.google.com/style/word-list#currently); (2) change "4" to "four", see [here](https://developers.google.com/style/numbers).

```suggestion
your own code. We support and document client libraries
for four main languages:
```

File: content/develop/connect/clients/_index.md [hunk 2]
```diff
@@ -15,20 +15,29 @@ title: Connect with Redis clients
...
+for these languages.
+
+We also develop libraries for the following languages but with limited
+support and documentation on external sites:
+
+- [Go](https://redis.uptrace.dev/guide/)
+- [Ruby](https://github.com/redis/redis-rb)
+- [C](https://github.com/redis/hiredis)
+
+To use the client libraries, you will need a Redis server to connect to.
```
Comment (dwdougherty) [inline content/develop/connect/clients/_index.md:36]:
```suggestion
You will need a Redis server to use these libraries.
```

File: content/develop/connect/clients/_index.md [hunk 3]
```diff
@@ -15,20 +15,29 @@ title: Connect with Redis clients
...
+You can experiment with a local installation of Redis Stack
+(see [Install Redis Stack]({{< relref "/operate/oss_and_stack/install/install-stack/" >}})) or with a free trial of [Redis Cloud]({{< relref "/operate/rc" >}}).
+To interact with a Redis server without writing code, use our
```
Comment (dwdougherty) [inline content/develop/connect/clients/_index.md:39]:
```suggestion
To interact with a Redis server without writing code, use
```

**Review by dwdougherty — CHANGES_REQUESTED** [review]
Some changes.

File: content/develop/connect/clients/_index.md (hunk 1 above)
Comment (andy-stark-redis) [reply content/develop/connect/clients/_index.md]: Done, and there was another "currently" elsewhere.

File: content/develop/connect/clients/_index.md (hunk 2 above)
Comment (andy-stark-redis) [reply content/develop/connect/clients/_index.md]: Done.

File: content/develop/connect/clients/_index.md (hunk 3 above)
Comment (andy-stark-redis) [reply content/develop/connect/clients/_index.md]: Done.

**Review by dwdougherty — APPROVED** [review]
LGTM

**Pull request merged**